src/NServiceBus.RabbitMQ.AcceptanceTests/ConfigureEndpointRabbitMQTransport.cs

```diff
@@ -57,19 +59,14 @@ public Task Configure(string endpointName, EndpointConfiguration configuration,
 
     ConnectionFactory CreateConnectionFactory()
     {
-        var match = Regex.Match(connectionString, string.Format("[^\\w]*{0}=(?<{0}>[^;]+)", "host"), RegexOptions.IgnoreCase);
-
-        var username = match.Groups["UserName"].Success ? match.Groups["UserName"].Value : "guest";
-        var password = match.Groups["Password"].Success ? match.Groups["Password"].Value : "guest";
-        var host = match.Groups["host"].Success ? match.Groups["host"].Value : "localhost";
-        var virtualHost = match.Groups["VirtualHost"].Success ? match.Groups["VirtualHost"].Value : "/";
+        var builder = new DbConnectionStringBuilder { ConnectionString = connectionString };
 
         var connectionFactory = new ConnectionFactory
         {
-            UserName = username,
-            Password = password,
-            VirtualHost = virtualHost,
-            HostName = host,
+            UserName = GetValueOrDefault(builder, "username", "guest"),
+            Password = GetValueOrDefault(builder, "password", "guest"),
+            VirtualHost = GetValueOrDefault(builder, "virtualhost", "nsb-rabbitmq-test"),
```

Comment:
All three spots that use this to set a default value have a problem. If a user has set a connection string in the environment variable that does not contain a VirtualHost, then the tests will run without one (the default of `/`, but the code using this `GetValueOrDefault` setting will use `nsb-rabbitmq-test` instead.

This means that queue purging would be run against a different virtual host than the tests ran on.